retrofit-adapters/rxjava2/src/test/java/retrofit2/adapter/rxjava2/CompletableTest.java

```diff
@@ -54,6 +54,22 @@
     observer.assertComplete();
   }
 
+  @Test public void completableSuccess204() {
+    server.enqueue(new MockResponse().setResponseCode(204));
+
+    RecordingCompletableObserver observer = observerRule.create();
+    service.completable().subscribe(observer);
+    observer.assertComplete();
```

Comment:
Is this a behavior change? What happened before the change?

Reply:
I don't believe so. This is the only case that sort of "safely" worked before because any onNext event in the underlying observable was knowingly ignored even if it was null (semantics of `ignoreElements()`). This worked before the change too, this is just to have a test for that behavior being explicit while other tests show it throws in non-Completable types now for 204/205 responses.

Other behaviors:
* Single and Maybe impls both silently just ignore nulls in onNext (Single throws NSEE, Maybe completes).
* Flowable and Observable both defer to the given observer, which may or may no nullcheck inputs (though presumably they'd be unexpected since RxJava 2 prohibits null items in the stream). They could complete if the onNext consumer handles/ignores nulls, or a stricter observer intermediary could fail early

Reply:
Right. I vaguely remember this being the observed behavior. Just wanted to make sure we don't deviate from it.